pkg/generate/source/detector.go

```diff
@@ -74,6 +75,11 @@ func DetectLiteralDotNet(dir string) *Info {
 	return detect(".net", dir, "project.json", "*.csproj")
 }
 
+// DetectGolang detects Go source
+func DetectGolang(dir string) *Info {
+	return detect("golang", dir, "main.go", "Godeps")
```

Comment:
Can we use *.go here, similar to *.csproj for .NET? At the moment s2i-go requires a main.go file, but I would like to fix that limitation eventually, to be able to run arbitrary programs (given a `go install`-style path)